Framework/DataHandling/src/LoadIsawDetCal.cpp

```diff
@@ -431,20 +431,23 @@ void LoadIsawDetCal::doRotation(V3D rX, V3D rY, DetectorInfo &detectorInfo,
   Quat Rot = Q2 * Q1;
 
   // Then find the corresponding relative position
-  const auto parent = comp->getParent();
-  if (parent) {
-    Quat rot0 = parent->getRelativeRot();
-    rot0.inverse();
-    Rot *= rot0;
-  }
-  const auto grandparent = parent->getParent();
-  if (grandparent) {
-    Quat rot0 = grandparent->getRelativeRot();
+  const auto componentIndex = componentInfo.indexOf(comp->getComponentID());
+
+  if (componentInfo.hasParent(componentIndex)) {
+    const auto parentIndex = componentInfo.parent(componentIndex);
```

Comment:
Maybe code like this gets more readable if we drop the `Index` everywhere?

```cpp
const auto parent = componentInfo.parent(component);
```

Unless we mix it with legacy-type pointers to `Component` it feels clear and natural to me. What do you think?